### ISSUE TITLE
"Deactivating an audio session that has running I/O"

### ISSUE THREAD

**Issue by bsudekum** [mapbox/mapbox-navigation-ios#1483]
Saw this in my logs today:

```
AVAudioSession.mm:1142:-[AVAudioSession setActive:withOptions:error:]: Deactivating an audio session that has running I/O. All I/O should be stopped or paused prior to deactivating the audio session.
```

https://stackoverflow.com/a/42433074/1522419 points out the answer pretty clearly, we need to stop the audioplayer before deactivating it.

This should happen here before unducking:

https://github.com/mapbox/mapbox-navigation-ios/blob/e1742ee166a71d469f3c10d0122147bb300746e4/MapboxNavigation/MapboxVoiceController.swift#L200-L204

We could potentially instead do this directly in unduckAudio before deactivating the player.

https://github.com/mapbox/mapbox-navigation-ios/blob/e1742ee166a71d469f3c10d0122147bb300746e4/MapboxNavigation/RouteVoiceController.swift#L214-L216

**Comment by tommyyoon** [comment]
I don't think you can just set it do false. What if my app is using audiossesion? Any suggestions on keeping the session active if my app is using audiosession?
